src/modules/uORB/uORBDevices.cpp

```diff
@@ -437,6 +437,7 @@ uORB::DeviceNode::publish(const orb_metadata *meta, orb_advert_t handle, const v
 	ret = devnode->write(nullptr, (const char *)data, meta->o_size);
 
 	if (ret < 0) {
+		errno = -ret;
```

Comment:
yes the convention is: <0 ret == error and errno is set to the error value, so that it can be compared, eg. with `errno == EAGAIN`.